application_sdk/clients/atlan_client.py

```diff
@@ -0,0 +1,64 @@
+import os
+from typing import Optional
+
+from pyatlan.client.atlan import AtlanClient
+
+from application_sdk.common.error_codes import ClientError
+from application_sdk.observability.logger_adaptor import get_logger
+
+logger = get_logger(__name__)
+
+
+def get_client(
+    base_url: Optional[str] = None,
+    api_key: Optional[str] = None,
+    api_token_guid: Optional[str] = None,
+) -> AtlanClient:
+    """
+    Returns an authenticated AtlanClient instance using provided parameters or environment variables.
+
+    Selects authentication method based on the presence of parameters or environment variables and validates the required configuration.
+    In general, the use of environment variables is recommended. Any parameters specified will override the environment variables.
+
+    Args:
+    base_url: Atlan base URL (overrides ATLAN_BASE_URL)
+    api_key: Atlan API key (overrides ATLAN_API_KEY)
+    api_token_guid: API token GUID (overrides API_TOKEN_GUID)
+    """
+    # Resolve final values (parameters override env vars)
+    final_token_guid = api_token_guid or os.environ.get("API_TOKEN_GUID")
+    final_base_url = base_url or os.environ.get("ATLAN_BASE_URL")
+    final_api_key = api_key or os.environ.get("ATLAN_API_KEY")
```

Comment:
1. can you move all the os.environ calls to `application_sdk/constants.py` under # Atlan Client? 
2. make then all prefix with `ATLAN_`